src/main/scala/com/softwaremill/mqperf/mq/KafkaMq.scala

```diff
@@ -70,43 +71,54 @@ class KafkaMq(configMap: Map[String, String]) extends Mq with StrictLogging {
       consumer.subscribe(List(Topic))
       consumer
     }
-    lazy val committerActor = {
-      val commitMs = configMap("commitMs").toLong.millis
-      system.actorOf(Props(new PeriodicKafkaCommitter(commitMs, consumer)))
-    }
 
-    var buffer: java.util.Iterator[ConsumerRecord[String, String]] = new java.util.ArrayList[ConsumerRecord[String, String]]().iterator()
-
-    override def receive(msgCount: Int): List[(MsgId, String)] = {
-      @tailrec
-      def nextMsg(): ConsumerRecord[String, String] = {
-        if (!buffer.hasNext) {
-          buffer = consumer.poll(pollTimeout.toMillis).iterator
-          nextMsg()
-        }
-        else
-          buffer.next()
+    def timeToCommit(): Boolean = System.nanoTime() - lastCommitTick > commitNs
+
+    def commitAsync(): Unit = {
+      logger.debug("Commit tick")
```

Comment:
We tick every n *seconds*, so in this case it's not relevant. I tried to put logging only in places like this. Otherwise using debug, because scalalogging uses macros,  so `logger.debug("x")` is becomes sth like `if (logger.isDebug) ....` to avoid unnecessary calls.